### DISCUSSION TITLE
Tópicos básicos para quem está começando como Front-End

### DISCUSSION THREAD

**Discussion by bomffas** [frontendbr/forum#616] [Discussão]
Boaa Galera,

Queria saber de vocês são os tópicos mais importantes para quem está começando no Front-End, esses tópicos são para uma talk. Eu pensei em algumas coisas:

- Semântica
- Acessibilidade
- Boas Práticas
- Comunidade ( Eventos e Open-Source )

**Comment by yanmagale** [comment]
@bfmjr Além dos  pontos que você já disse, focar sempre na base: HTML, CSS e JS.

**Comment by munizart** [comment]
Opa, boa lista.
Acho interessante também versionamento, com git, e JavaScript (vanilla, com prototypical inheritance e FP)

**Comment by deppbrazil** [comment]
Hoje a base do front é o html css e js, como estas 3 linguagens estão em constante crescimento vale lembrar que para quem iniciar no mundo front é sempre importante que aprendam a desenvolver também nativamente antes de começarem a usar os frameworks e libs, pois assim vão entender o que estão fazendo, e conseguir identificar muitos problemas no código, caso seja necessário... pois o que vejo de pessoas desenvolvendo utilizando por exemplo o bootstrap, mas não sabem fazer um `@media querie ` em css puro. Buscar entender a melhoria em tags no html5 é muito importante, e o que o css3 oferece na folha de estilo e cair de cabeça no js nativo antes de começar a se aventurar no react, angular etc... 

Os frameworks e libs vão te ajudar muito, mas é sempre bom dar uma treinada no desenvolvimento nativo. =)

**Comment by felipechalreo** [comment]
Inglês.

**Comment by felipefialho** [comment]
- HTML
- CSS
- JavaScript

**Comment by devigor** [comment]
- HTML
- CSS
- JavaScript
- Inglês (O suficiente para saber ler artigos)
- Força de vontade

**Comment by mahenrique94** [comment]
- HTML5
- CSS3
- Javascript
- Inglês
- Noção de design
- Design UX
- Acessibilidade

**Comment by nandomoreirame** [comment]
Eu colocaria Noção de design, UX e Acessibilidade como tópicos para intermediário @mahenrique94 
Saber a base: HTML, CSS e JS para quem tá começando já é o sufissiente :smile:

**Comment by willycamargo** [comment]
Eu arriscaria dizer apenas HTML, CSS e JS.

E ainda diria para não se assustar com quem recomendar para estudar além disso. 

Focar em conteúdos que se propõe a ensinar apenas os fundamentos e com o passar dos dias os problemas irão aparecer e irão levar a estudar todo o resto que for necessário.


Um problema grave e que faz alguns deixar de estudar nossa área é a quantidade de conteúdo que as pessoas exigem que precisam saber, sendo que na verdade HTML, CSS e JS já é suficiente para desenvolver muita coisa boa.

**Comment by pedrobritto** [comment]
Eu diria que tentar enxergar o objetivo do produto que está sendo desenvolvido, não só as ferramentas, principalmente se o desenvolvedor tiver freela em mente ou muita liberdade para criar. Pra sempre ter em mente que HTML, CSS e JS são ferramentas para um fim, e não deixar esse fim de lado.

Exemplo: Desenvolver um site que converta mais clientes para uma floricultura local é o fim, desenvolver componentes é o meio.

**Comment by woliveiras** [comment]
@bfmjr, eu listei algumas coisas aqui: https://github.com/woliveiras/front-end-career

**Comment by phenriqueleao** [comment]
> Eu arriscaria dizer apenas HTML, CSS e JS.
E ainda diria para não se assustar com quem recomendar para estudar além disso.

Muito bem observado, levando em consideração que a pessoa é uma iniciante total. O ideal é tentar resumir ao máximo o número de coisas que ela precisa aprender. Nada me assustava mais do que aquelas listas com 30 itens que eu precisaria saber para começar no front-end.
